tools/dissector/sample_manager.cpp

```diff
@@ -328,51 +333,49 @@ namespace OpenDDS
       }
 
       if (no_dcps_data_types) {
-        ACE_DEBUG((LM_DEBUG,
+        ACE_DEBUG((LM_WARNING,
                    ACE_TEXT(
                      "%C has no types defined as a primary DCPS data types.\n"),
                    filename));
         primary_dissectors = dissectors_; // Evaluate All Dissectors
       }
 
       // Build WS Namespace and Fields
-      for (
-        DissectorsType::iterator i = primary_dissectors.begin();
-        i != primary_dissectors.end();
-        i++
-      ) {
-        // Get Namespaces from name
-        Sample_Base::clear_ns();
-        std::string name(i->first);
-
-        // If name begins with "IDL:", remove it
-        const std::string idl_prefix("IDL:");
-        if (name.substr(0, idl_prefix.size()) == idl_prefix) {
-            name.erase(0, idl_prefix.size());
-        }
-
-        // If name contains ':', remove everything after the last ':'
-        size_t l = name.rfind(":");
-        if (l != std::string::npos) {
-            name.erase(l, name.size() - l);
+      try {
+        // First Pass
+        for (
+          DissectorsType::iterator i = primary_dissectors.begin();
+          i != primary_dissectors.end();
+          i++
+        ) {
+          // Get Namespace from name
+          Sample_Base::push_idl_name(i->first);
+
+          // Create WS Fields off this namespace location
+          Sample_Dissector & dissector = *i->second;
+          dissector.init_ws_proto_tree(true); // first pass, evaluate tree
+          Sample_Base::clear_ns();
         }
 
-        // Push namespace when we find a '/'
-        name.push_back('/'); // For last namespace
-        l = 0;
-        for (size_t i = 0; i != name.size(); i++) {
-            if (name[i] == '/') {
-                Sample_Base::push_ns(name.substr(i - l, l));
-                l = 0;
-            } else {
-                l++;
-            }
+        // Second Pass
+        for (
+          DissectorsType::iterator i = primary_dissectors.begin();
+          i != primary_dissectors.end();
+          i++
+        ) {
+          // Get Namespace from name
+          Sample_Base::push_idl_name(i->first);
+
+          // second pass, collect fields to pass to Wireshark
+          i->second->init_ws_proto_tree(false);
+          Sample_Base::clear_ns();
         }
 
-        // Create WS Fields off this namespace location
-        Sample_Dissector & dissector = *i->second;
-        dissector.init_ws_proto_tree();
-        Sample_Base::clear_ns();
+      } catch (const Sample_Dissector_Error & e) {
+        ACE_DEBUG((LM_ERROR,
+          ACE_TEXT("Error during dissector initization of %s: %s\n"),
```

Comment:
%C is for char

Reply:
Adam has told that before, but it's a hard habit to break.